lib/get-esm-exports.js

```diff
@@ -1,95 +1,200 @@
+const { importAssertions } = require('acorn-import-assertions')
+const warn = require('./helpers')
+
+const TS_EXTENSION_RE = /\.(ts|mts|cts)$/
 
 const acornOpts = {
   ecmaVersion: 'latest',
   sourceType: 'module'
 }
 
 const parser = Parser.extend(importAssertions)
+const FILE_NAME = 'get-esm-exports'
 
-function warn (txt) {
-  process.emitWarning(txt, 'get-esm-exports')
-}
+function getEsmExports(moduleStr, generate=false, url=undefined) {
+  const exportSpecifierNames = new Set()
+  const exportAlias = {}
+  let ast
+  
+  // if it's a typescript file, we need to parse it with recasts typescript parser
+  if (url && TS_EXTENSION_RE.test(url)) {
+    ast = recast.parse(moduleStr, {parser: require("recast/parsers/typescript")})
+  } else {
+    ast = recast.parse(moduleStr, {parser: {
+      parse(source) {
+        return parser.parse(source, acornOpts)
+      }
+    }})
+  }
 
-function getEsmExports (moduleStr) {
-  const exportedNames = new Set()
-  const tree = parser.parse(moduleStr, acornOpts)
-  for (const node of tree.body) {
-    if (!node.type.startsWith('Export')) continue
-    switch (node.type) {
-      case 'ExportNamedDeclaration':
-        if (node.declaration) {
-          parseDeclaration(node, exportedNames)
-        } else {
-          parseSpecifiers(node, exportedNames)
+  recast.visit(ast, {
+    visitExportNamedDeclaration(path) {
+      const node = path.node
+  
+      if (node.declaration) {
+        parseDeclaration(node.declaration, exportAlias)
+      } else {
+        parseSpecifiers(node.specifiers, exportAlias, exportSpecifierNames)
+      }
+  
+      this.traverse(path)
+    },
+  
+    visitExportDefaultDeclaration(path) {
+      const node = path.node
+      const iitmRenamedExport = 'iitmRenamedExport'
+  
+      if (['ObjectExpression', 'ArrayExpression', 'Literal'].includes(node.declaration.type) && generate) {
+        const variableDeclaration = {
+          type: 'VariableDeclaration',
+          kind: 'let',
+          declarations: [
+            {
+              type: 'VariableDeclarator',
+              id: {
+                type: 'Identifier',
+                name: iitmRenamedExport,
+              },
+              init: node.declaration,
+            },
+          ],
         }
-        break
-      case 'ExportDefaultDeclaration':
-        exportedNames.add('default')
-        break
-      case 'ExportAllDeclaration':
-        if (node.exported) {
-          exportedNames.add(node.exported.name)
-        } else {
-          exportedNames.add('*')
+        path.replace(variableDeclaration)
+  
+        const newExportDefaultDeclaration = {
+          type: 'ExportDefaultDeclaration',
+          declaration: {
+            type: 'Identifier',
+            name: iitmRenamedExport,
+          },
         }
-        break
-      default:
-        warn('unrecognized export type: ' + node.type)
+        path.insertAfter(newExportDefaultDeclaration)
+      } else {
```

Comment:
Are you certain this branch is and will continue to be valid for _all_ exceptions to the `if` condition above? Would this be safer as an `else if (condition)`?

Reply:
I would assume so because I have added a test for all the possible default exports based off of this:

https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export

Reply:
but it might be safer to do an else if